blueprints/ember-bootstrap/index.js

```diff
@@ -1,241 +1,333 @@
+  'bootstrapVersion',
+  'importBootstrapTheme',
+  'importBootstrapCSS',
+  'importBootstrapFont',
+  'insertEmberWormholeElementToDom',
+  'whitelist'
+];
 
-const bs3Version = '^3.3.7';
-const bs4Version = '^4.0.0';
+function readConfig() {
+  let configPath = resolve(BUILD_FILE);
 
-const validPreprocessors = [
-  'none',
-  'less',
-  'sass'
-];
+  return ConfigBuilder.create(configPath).then(configBuilder => {
+    let addonConfig = configBuilder.get(GEN_NAME);
+    addonConfig = ConfigBuilder.parse(addonConfig);
+    if (!addonConfig) {
+      addonConfig = {};
+    }
+
+    return { addonConfig, configBuilder };
+  });
+}
+
+function outputConfig(ui, config) {
+  config = ConfigBuilder.stringify(config, null, '  ');
+  ui.writeLine(
+    chalk.red(
+      `Configuration file could not be edited. Manually update your ${BUILD_FILE} to include '${GEN_NAME}': ${config}`
+    )
+  );
+}
+
+function normalizeOptions(options) {
+  if (wizard.BS_VERSIONS.indexOf(options.bootstrapVersion) === -1) {
+    options.bootstrapVersion = null;
+  }
+
+  if (wizard.PREPROCESSORS.indexOf(options.preprocessor) === -1) {
+    options.preprocessor = null;
+  }
+
+  if (options.bootstrapVersion === 4 && options.preprocessor === 'less') {
+    options.preprocessor = null;
+  }
+}
+
+function normalizeConfig(config) {
+  return Object.keys(config).reduce((acc, key) => {
+    if (AVAILABLE_OPTIONS.indexOf(key) === -1) {
+      return acc;
+    }
+    acc[key] = config[key];
+    return acc;
+  }, {});
+}
+
+function stringifyConfig(config) {
+  return Object.keys(config).reduce((acc, key) => {
+    acc[`${GEN_NAME}.${key}`] = ConfigBuilder.stringify(
+      config[key],
+      null,
+      '  '
+    );
+    return acc;
+  }, {});
+}
 
 module.exports = {
-  name: 'ember-bootstrap',
+  name: GEN_NAME,
 
   description: 'Configure ember-bootstrap',
 
+  works: 'insideProject',
+
   availableOptions: [
     { name: 'bootstrap-version', type: Number, aliases: ['bootstrap', 'bv'] },
     { name: 'preprocessor', type: String, aliases: ['pp'] }
   ],
 
-  works: 'insideProject',
-
-  normalizeEntityName() {
+  normalizeEntityName() {},
+
+  addonConfig: {},
+  configBuilder: null,
+
+  beforeInstall(options) {
+    normalizeOptions(options);
+    // try to read config
+    return (
+      readConfig()
+        .then(config => {
+          // save config builder to use later
+          this.configBuilder = config.configBuilder;
+          // and pass addon config to the next chain
+          return config.addonConfig;
+        })
+        // use empty object in case there are no way to read the current config
+        .catch(() => ({}))
+        .then(config => {
+          config.bootstrapVersion = this._detectBootstrapVersion(
+            options,
+            config
+          );
+          config.preprocessor = this._detectPreprocessor(options);
+          this.addonConfig = config;
+          return this.ui.prompt(wizard.questions(options, config));
```

Comment:
We talked about the issue that FastBoot tests run the default blueprint, though there is currently no way to answer any prompts programmatically. It seems the prompts show up without any escape hatch here (like an env var to suppress them). How is this working now? (i.e. why where the tests even passing :))